hephaistus/main-theme/distribution-system/chord/playground/nullist0/lib/src/main/kotlin/org/lia/dht/nullist0/chord/model/ChordScope.kt

```diff
@@ -0,0 +1,16 @@
+package org.lia.dht.nullist0.chord.model
+
+/**
+ * An interface to build identifier of Chord Ring.
+ */
+interface ChordScope<Id, Value> {
+    fun AbstractChordNode<Id, Value>.toId(): Id
+
+    fun bitSize(): Int
+
+    fun Int.toId(): Id
+
+    infix operator fun Id.plus(other: Id): Id
+
+    infix operator fun Id.compareTo(other: Id): Int
```

Comment:
Chord is ring type DHT, so it should be able to find the beginning and end of the ring.

Reply:
Thanks for comment. I added some comments to the `compareTo` and `plus` because they may different from natural integers. Also, the implementation of interval function is moved too, because it will expose the notion of Chord ring.